File: Back-end/bidhouse-common/src/main/java/com/bidhouse/models/House.java
package com.bidhouse.models;

import java.util.Date;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import com.bidhouse.annotation.CascadeSave;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "houses")
public class House {
	@Id
	private String id;
	private String address;
	private String owner;
	private double area;
	private double startPrice;
	private double endPrice;
	private boolean sessionEnd;
	private Date createdDate;
	private Date endDate;
	@DBRef(db = "bid_states")
	@CascadeSave
	private List<BidState> bidStates;
	
}

File: Front-end/src/shared/models/house.ts
export class House {
   
    public _id : string;
    public address: string;
    public area : number;
    public sessionEnd : boolean;
    public bidStates : Array<object>;
    public createdDate : Date;
    public endDate : Date;
    public timeLeft : Date;
    public displayTimeLeft : any;
    public startPrice : number;
    public endPrice : number;
    public owner : string;
    public info : object;
    public type : string;
    public avatar : string;


    constructor(_id? : string, address? : string, area? : number, sessionEnd? : boolean, bidStates? : Array<object>, createdDate? : string, endDate? : string, startPrice? : number, endPrice? : number, owner? : string) {
        this._id = "";
        this.address = "";
        this.area = 0;
        this.sessionEnd = false;
        this.bidStates = [];
        this.createdDate = new Date();
        this.endDate = new Date();
        this.timeLeft = new Date(+this.endDate - +this.createdDate);
        this.displayTimeLeft = [];
        this.startPrice = 0;
        this.endPrice = 0;
        this.owner = "";
        this.info = {
            beds : 0,
            baths : 0,
            sqMeter : 0
        };
        this.type = "";
        this.avatar = "./assets/images/no-image.jpg";
    }
} 